lib/src/kbasesearchengine/system/TypeStorage.java

```diff
@@ -2,16 +2,37 @@
 
 import java.util.List;
 
-import kbasesearchengine.events.exceptions.IndexingException;
-
+/** Stores type documents that describe how to transform an object from a a data source into
+ * one or more documents that can be indexed by search and mappings from data source type versions
+ * to search transformation document versions.
+ * @author gaprice@lbl.gov
+ *
```

Comment:
Provides access to a list of available ObjectTypeParsingRules that may be filtered by StorageObjectType?

Reply:
Clarified